### PR TITLE
kommander: enable federation of kubeaddons controller

### PR DESCRIPTION
**Pull request by alejandroEsc**
change allow federating kubeaddons controller

Tested locally, we assume by default we are deploying on a konvoy cluster, to avoid deleting the kubeaddons namespace when removing the chart.

### CODE REVIEW

File: stable/kommander/values.yaml [hunk 1]
```diff
@@ -53,9 +53,12 @@ federate:
   clusteraddons: true
   clusterrolebindings: true
   customresourcedefinitions: true
+  kubeaddons: true
   roles: true
   rolebindings: true
 
+  # set to true if you wish to deploy on a non konvoy cluster
```
Comment (jimmidyson) [inline stable/kommander/values.yaml]: My one concern is that now we need to know whether we're deploying on Konvoy or non-Konvoy and can't use this as same resource(s) to federate to all joined clusters. Is there a way round this? As in, perhaps a pre-install hook that only creates kubeaddons namespace if required?

Comment (alejandroEsc) [reply stable/kommander/values.yaml]: alright, wasnt sure how we wanted to handle that. I will test a bit with a pre-install hook and get back to you. The word is that helm should not delete the resource in this case, which is what i would expect, but let me verify.

Comment (s12chung) [reply stable/kommander/values.yaml]: could add it as an init container for kommander?

Comment (alejandroEsc) [reply stable/kommander/values.yaml]: @s12chung yep, digging into that possibility now.

File: stable/kommander/values.yaml [hunk 2]
```diff
@@ -53,9 +53,15 @@ federate:
   clusteraddons: true
   clusterrolebindings: true
   customresourcedefinitions: true
+  kubeaddons: true
   roles: true
   rolebindings: true
 
+  image:
+    repository: "mesosphere/kubeaddons-addon-initializer"
```
[the remark this answers is not in the document]
Comment (alejandroEsc) [reply stable/kommander/values.yaml]: This is the image that is responsible for creating the namespace if necessary.

File: stable/kommander/values.yaml [hunk 3]
```diff
@@ -53,9 +53,15 @@ federate:
   clusteraddons: true
   clusterrolebindings: true
   customresourcedefinitions: true
+  kubeaddons: true
   roles: true
   rolebindings: true
 
+  image:
```
[the remark this answers is not in the document]
Comment (alejandroEsc) [reply stable/kommander/values.yaml]: image is what we name docker images  normally.

**Pull request merged**